When folding tuples the trailing comma is not removed

## Configuration ##
```
version = 3.7.14
runner.dialect = scala213
newlines.source = fold
```

## Command-line parameters ##
```
$ scalafmt --version
scalafmt 3.7.14

$ scalafmt src

$ scalafmt src
/(...)/src/Main.scala:2: error: [dialect scala213] illegal start of simple expression
  val x = ("a", "b", )
                     ^
```

## Steps

Given code like this:
```scala
object Main {
  val x = (
    "a", 
    "b",
  )

  val y = Seq(
    "a",
    "b",
  )
}
```

## Problem

Scalafmt formats code like this:
```scala
object Main {
  val x = ("a", "b", )

  val y = Seq("a", "b")
}
```

## Expectation

I would like the formatted output to look like this:
```scala
object Main {
  val x = ("a", "b")

  val y = Seq("a", "b")
}
```

## Notes
I also tried with `rewrite.trailingCommas.allowFolding = false` and this does not help.

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RewriteTrailingCommas.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RewriteTrailingCommas.scala
@@ -42,7 +42,10 @@ private class RewriteTrailingCommas(ftoks: FormatTokens)
 
       // comma and paren/bracket/brace need to have the same owner
       (rightOwner eq nft.meta.rightOwner) && (nft.right match {
-        case _: Token.RightBracket | _: Token.RightParen =>
+        case _: Token.RightParen =>
+          rightOwner.isAny[Member.SyntaxValuesClause, Member.Tuple]
+
+        case _: Token.RightBracket =>
           rightOwner.is[Member.SyntaxValuesClause]
 
         case _: Token.RightBrace =>

--- a/scalafmt-tests/src/test/resources/trailing-commas/trailingCommasAlwaysSingleLine.stat
+++ b/scalafmt-tests/src/test/resources/trailing-commas/trailingCommasAlwaysSingleLine.stat
@@ -21,6 +21,4 @@ val x = (
   "b",
 )
 >>>
-test does not parse
-val x = ("a", "b", )
-                   ^
+val x = ("a", "b")

--- a/scalafmt-tests/src/test/resources/trailing-commas/trailingCommasAlwaysSingleLine.stat
+++ b/scalafmt-tests/src/test/resources/trailing-commas/trailingCommasAlwaysSingleLine.stat
@@ -15,3 +15,12 @@ foo(
 )
 >>>
 foo(a)
+<<< #3663 tuple
+val x = (
+  "a", 
+  "b",
+)
+>>>
+test does not parse
+val x = ("a", "b", )
+                   ^

--- a/scalafmt-tests/src/test/resources/trailing-commas/trailingCommasNeverSingleLine.stat
+++ b/scalafmt-tests/src/test/resources/trailing-commas/trailingCommasNeverSingleLine.stat
@@ -21,6 +21,4 @@ val x = (
   "b",
 )
 >>>
-test does not parse
-val x = ("a", "b", )
-                   ^
+val x = ("a", "b")

--- a/scalafmt-tests/src/test/resources/trailing-commas/trailingCommasNeverSingleLine.stat
+++ b/scalafmt-tests/src/test/resources/trailing-commas/trailingCommasNeverSingleLine.stat
@@ -15,3 +15,12 @@ foo(
 )
 >>>
 foo(a)
+<<< #3663 tuple
+val x = (
+  "a", 
+  "b",
+)
+>>>
+test does not parse
+val x = ("a", "b", )
+                   ^

--- a/scalafmt-tests/src/test/resources/trailing-commas/trailingCommasPreserve.stat
+++ b/scalafmt-tests/src/test/resources/trailing-commas/trailingCommasPreserve.stat
@@ -353,3 +353,13 @@ object a {
       b: Int /* c */
   ) = a + b
 }
+<<< #3663 tuple
+val x = (
+  "a", 
+  "b",
+)
+>>>
+val x = (
+  "a",
+  "b",
+)

